docs/root/faq/configuration/timeouts.rst

```diff
@@ -53,6 +53,14 @@ context request/stream is interchangeable.
   is the amount of time that the connection manager will allow a stream to exist with no upstream
   or downstream activity. The default stream idle timeout is *5 minutes*. This timeout is strongly
   recommended for streaming APIs (requests or responses that never end).
+* The HTTP protocol :ref:`max_stream_duration <envoy_api_field_core.HttpProtocolOptions.max_stream_duration>` 
```

Comment:
As implemented right now this only works for downstream connection configuration, so I would clarify that.

Reply:
We should move it to DownstreamCommonHttpProtocolOptions, shouldn't we? And, It is not required to add this functionality to upstream to resolve the proposed issue but is required in the future?

Reply:
The only case that I think we would ever need this upstream is for local origin (async client) requests. Otherwise by definition this should apply to both upstream/downstream at the same time. If you think we would want this for upstream/async client eventually we can keep it on the base message. If you think that we should only have it downstream we should move it. I guess for future proofing maybe I would leave it and just clarify that it only applies downstream currently? WDYT?

Reply:
Ok, for now. we should leave it on the current message. I will specify it as todo and apply it to upstream in the future. thank you for your comments.